dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsInterface.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *    Copyright 2018 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.webservice.permissions;
+
+import java.util.List;
+
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.Workflow;
+
+/**
+ * <p>
+ * Abstracts out the backend permissions service.
+ * </p>
+ * <p>
+ * A <code>Permission</code> is an email address
+ * and a  {@link Role},
+ * an enum whose values are <code>OWNER</code>,
+ * <code>WRITER</code>, and <code>READER</code>.
+ * </p>
+ * <p>
+ * An email address can be for a single user or for a group.
+ * </p>
+ * <p>
+ * An <code>Workflow</code> can have multiple <code>Permission</code>s associated
+ * with it.
+ * </p>
+ *
+ * <p>
+ *     If we extend sharing to tools, should we create additional methods for tools here,
+ *     e.g., <code>setPermission(Tool tool, User requester, Permission permission)</code>,
+ *     or should we instead make this interface more generic, and just require a
```

Comment:
(change to comment) Probably just make this generic in the future:
i.e. `List<Permission> setPermission(Entry entry, User requester, Permission permission);` instead of `List<Permission> setPermission(Workflow workflow, User requester, Permission permission);`